hibernate-core/src/main/java/org/hibernate/query/sqm/internal/SqmJdbcExecutionContextAdapter.java

```diff
@@ -70,4 +70,8 @@ public boolean hasQueryExecutionToBeAddedToStatistics() {
 		return true;
 	}
 
+	@Override
+	public boolean upgradeLocks() {
+		return true;
```

Comment:
This means that only HQL/Criteria and native queries can upgrade locks. Is that intentional i.e. did we not upgrade locks for `session.find()` before?